packages/gatsby-plugin-page-creator/src/babel-remove-collection-graphql.ts

```diff
@@ -0,0 +1,171 @@
+import { NodePath, PluginObj } from "@babel/core"
+import { Binding } from "@babel/traverse"
+import {
+  CallExpression,
+  Expression,
+  Identifier,
+  ImportDeclaration,
+  ObjectExpression,
+  Program,
+  StringLiteral,
+  TaggedTemplateExpression,
+} from "@babel/types"
+
+interface IGraphQLTag {
+  isGlobal: boolean
+}
+
+const isGlobalIdentifier = (
+  tag: NodePath,
+  tagName: string = `graphql`
+): boolean =>
+  tag.isIdentifier({ name: tagName }) && tag.scope.hasGlobal(tagName)
+
+export function followVariableDeclarations(binding: Binding): Binding {
+  const node = binding.path?.node
+  if (
+    node?.type === `VariableDeclarator` &&
+    node?.id.type === `Identifier` &&
+    node?.init?.type === `Identifier`
+  ) {
+    return followVariableDeclarations(
+      binding.path.scope.getBinding(node.init.name) as Binding
+    )
+  }
+  return binding
+}
+
+function getTagImport(tag: NodePath<Identifier>): NodePath | null {
+  const name = tag.node.name
+  const binding = tag.scope.getBinding(name)
+
+  if (!binding) return null
+
+  const path = binding.path
+  const parent = path.parentPath
+
+  if (
+    binding.kind === `module` &&
+    parent.isImportDeclaration() &&
+    parent.node.source.value === `gatsby`
+  )
+    return path
+
+  if (
+    path.isVariableDeclarator() &&
+    (path.get(`init`) as NodePath).isCallExpression() &&
+    (path.get(`init.callee`) as NodePath).isIdentifier({ name: `require` }) &&
+    ((path.get(`init`) as NodePath<CallExpression>).node
+      .arguments[0] as StringLiteral).value === `gatsby`
```

Comment:
It may be a template literal for `require` (it would be in our code). Don't think this matters much here.